### ISSUE TITLE
Slowdown of variant effect prediction with `torch.backends.cudnn.deterministic = True`

### ISSUE THREAD

**Issue by kathyxchen** [FunctionLab/selene#180]
The recent addition of 

```
torch.backends.cudnn.deterministic = True
torch.backends.cudnn.benchmark = False
```
has led to a 10x slowdown of variant effect prediction (e.g. taking 1000s to process 10k variants when previously it took 100s to process 10k variants) for one of my DL models. I've tested that this is the reason for the slowdown by running the same script commenting in/out these 2 lines. 

I'm wondering if this is because the model was not trained with these parameters, since I would hope this slowdown doesn't happen for models that are trained with `deterministic=True`?  @rfriedman22 would love to get your input. I'm also going to try training some models with/without the random seed to see if I can figure out what's going on and how we can incorporate a workaround.

**Comment by rfriedman22** [comment]
What if we moved those two lines to somewhere within the `TrainModel` class? I would think that seeding random number generators, and setting Pytorch to be deterministic, would not matter for making predictions since the model is set and static at that point anyways?

If it would help, I can also try making predictions on my model trained with `deterministic=True` with/out these two lines to see how it affects runtime.

**Comment by kathyxchen** [comment]
Yup that's what I was thinking, it seems strange that it would impact the prediction component at all... 

I was originally thinking I could modify to check if `train` is set in the config file, but moving it to TrainModel class is a better idea in case other people are using Selene through the API - I'll make that change :) 

But yeah I am curious if you have noticed any slowdown in training when these are set? I'll also be training a couple models soon so no worries if you haven't / no need to go out of your way to do that :+1:

**Comment by rfriedman22** [comment]
It is strange indeed...no idea why that should happen.

I was thinking about checking if `train` is set in the config too, but it would still create problems if the config had e.g. both `train` and `eval`. So between that, and the fact that people might use Selene via the API instead of the CLI, I agree we should compartmentalize the lines to the class.

I haven't really noticed a slowdown since I added these lines. It's possible it is slower during training, but if so it isn't a 10x slowdown or I would have noticed.

FWIW the [page](https://pytorch.org/docs/stable/notes/randomness.html) on this does warn that determinism can cause slowdowns, but it's still not clear to me why that should appear at test time.
